text/0148-sampling-probability.md

```diff
@@ -0,0 +1,744 @@
+      - [Dapper's "Inflationary" Sampler](#dappers-inflationary-sampler)
+  * [Working with adjusted counts](#working-with-adjusted-counts)
+    + [Merging samples](#merging-samples)
+    + [Maintaining "Probability proportional to size"](#maintaining-probability-proportional-to-size)
+    + [Zero adjusted count](#zero-adjusted-count)
+- [Proposed Tracing specification](#proposed-tracing-specification)
+  * [Suggested text](#suggested-text)
+- [Recommended reading](#recommended-reading)
+- [Acknowledgements](#acknowledgements)
+
+<!-- tocstop -->
+
+Objective: Specify a foundation for sampling techniques in OpenTelemetry.
+
+## Motivation
+
+Probability sampling allows consumers of sampled telemetry data to
+collect a fraction of telemetry events and use them to estimate total
+quantities about the population of events, such as the total rate of
+events with a particular attribute.  Sampling is a general-purpose
+facility for lowering cost at the expense of lower data quality.
+
+These techniques enable reducing the cost of telemetry collection,
+both for producers (i.e., SDKs) and for processors (i.e., Collectors),
+without losing the ability to (at least coarsely) monitor the whole
+system.
+
+Sampling builds on results from probability theory, most significantly
+the concept of expected value.  Estimates drawn from probability
+samples are *random variables* that, when correct procedures are
+followed, accurately reflect their true value, making them unbiased.
+Unbiased samples can be used for after-the-fact analysis.  We can
+answer questions such as "what fraction of events had property X?"
+using the fraction of events in the sample that have property X.
+
+This document outlines how producers and consumers of sample telemetry
+data can convey estimates about the total count of telemetry events,
+without conveying information about how the sample was computed, using
+a quantity known as **adjusted count**.  In common language, a
+"one-in-N" sampling scheme emits events with adjusted count equal to
+N.  Adjusted count is the expected value of the number of events in
+the population represented by an individual sample event.
+
+## Examples
+
+In all of these examples, the use of probability sampling leads to an
+attribute named like `sampling.sampler_name.adjusted_count`.
+Consumers of spans, metrics, and logs annotated with adjusted counts
+are able to calculate accurate statistics about the population of
+events, at a basic level, without knowing details about the sampling
+configuration.
+
+### Span sampling
+
+<details> 
+<summary> Example use-cases for probability sampling of spans
+generally involve generating metrics from spans.</summary>
+
+#### Sample spans to Counter Metric
+
+For every span it receives, the example processor will synthesize
+metric data as though a Counter instrument named `S.count` for span
+named `S` had been incremented once per span at the original `Start()`
```

Comment:
One potential challenge here is that the span `Start()` will not have access to all the attributes (e.g. some of the attributes might be added later).

Reply:
I see now the consequences of moving the examples above the introductory text, some missing bits. I've added "complete" here to avoid the ambiguity. I'm trying to stick the OTel tracing data model, where spans encoded in OTLP are already finished.